### PR TITLE
Update to latest core v7

### PR DESCRIPTION
**Pull request by andreasohlund**

### CODE REVIEW

File: src/NServiceBus.Callbacks.AcceptanceTests/EndpointTemplates/DefaultServer.cs [hunk 1]
```diff
@@ -22,42 +19,40 @@ public DefaultServer(List<Type> typesToInclude)
...
-                configuration.UseSerialization((SerializationDefinition) Activator.CreateInstance(serializerType));
-            }
-            await configuration.DefinePersistence(settings, endpointConfiguration).ConfigureAwait(false);
+            configuration.RegisterComponentsAndInheritanceHierarchy(runDescriptor);
 
-            configuration.GetSettings().SetDefault("ScaleOut.UseSingleBrokerQueue", true);
             configurationBuilderCustomization(configuration);
 
-            return configuration;
+            //runDescriptor.OnTestCompleted(rst =>
```
Comment (timbussmann) [inline src/NServiceBus.Callbacks.AcceptanceTests/EndpointTemplates/DefaultServer.cs]: 🔥

File: src/NServiceBus.Callbacks.AcceptanceTests/EndpointTemplates/TestHelperExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    public static class TestHelperExtensions
+    {
+
+        public static void RegisterComponentsAndInheritanceHierarchy(this EndpointConfiguration builder, RunDescriptor runDescriptor)
+        {
+            builder.RegisterComponents(r => { RegisterInheritanceHierarchyOfContextOnContainer(runDescriptor, r); });
+        }
+
+
+        static void RegisterInheritanceHierarchyOfContextOnContainer(RunDescriptor runDescriptor, IConfigureComponents r)
```
Comment (timbussmann) [inline src/NServiceBus.Callbacks.AcceptanceTests/EndpointTemplates/TestHelperExtensions.cs]: so this isn't contained in the att framework? should this be moved from the acceptance tests to the framework ?

**Comment by andreasohlund** [comment]
@timbussmann should be good to go

File: src/NServiceBus.Callbacks/NServiceBus.Callbacks.csproj [hunk 3]
```diff
@@ -1,25 +1,22 @@
...
     <NoWarn>CS0419</NoWarn>
-    <UpdateAssemblyInfo>true</UpdateAssemblyInfo>
+    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
+    <Description>NServiceBus Callbacks</Description>
   </PropertyGroup>
+  
   <ItemGroup>
-    <PackageReference Include="NServiceBus" Version="6.0.0" />
+    <PackageReference Include="NServiceBus" Version="7.0.0-alpharelease0051" />
     <PackageReference Include="Particular.CodeRules" Version="0.2.0" />
```
Comment (timbussmann) [inline src/NServiceBus.Callbacks/NServiceBus.Callbacks.csproj]: should have `PrivateAssets="All"`

File: src/NServiceBus.Callbacks/NServiceBus.Callbacks.csproj [hunk 4]
```diff
@@ -1,25 +1,22 @@
...
-    <PackageReference Include="NuGetPackager" Version="0.6.3" />
     <PackageReference Include="GitVersionTask" Version="4.0.0-beta0012" />
```
Comment (timbussmann) [inline src/NServiceBus.Callbacks/NServiceBus.Callbacks.csproj]: probably easier to use https://github.com/Particular/NServiceBus/blob/develop/src/Directory.Build.props directly?

File: src/NServiceBus.Callbacks/NServiceBus.Callbacks.csproj (hunk 3 above)
Comment (timbussmann) [inline src/NServiceBus.Callbacks/NServiceBus.Callbacks.csproj]: same in the callbacks.testing package btw.

File: src/NServiceBus.Callbacks/NServiceBus.Callbacks.csproj (hunk 4 above)
Comment (andreasohlund) [inline src/NServiceBus.Callbacks/NServiceBus.Callbacks.csproj]: Not following?

Comment (andreasohlund) [inline src/NServiceBus.Callbacks/NServiceBus.Callbacks.csproj]: So you mean that we pull 


```
    <PackageReference Include="Particular.CodeRules" Version="0.2.0" />
    <PackageReference Include="GitVersionTask" Version="4.0.0-beta0012" />
```

up to the build props to have the applies to all projects?

Makes sense I guess

**Comment by andreasohlund** [comment]
@timbussmann addressed your comments

**Pull request merged**